src/components/fields/DateField.js

```diff
@@ -42,40 +40,124 @@ class DateField extends AskField {
     )
   }
 
-  validateAndSave (options) {
-    this.validate()
-    this.update(options)
+  validateAndSave(options) {
+    if (this.validate()) {
+      this.setState({ value: this.buildValue() });
+    } else {
+      this.setState({ value: '--' });
+    }
+    this.update(options);
   }
 
   // Interface methods
 
-  validate () {
-    let isValid = true
-    let isCompleted = false
-    isCompleted = !!this.state.value.length
-    this.setState({ isValid: isValid, completed: isCompleted })
-    return this.props.wrapper.required ? isValid && isCompleted : isValid
+  validate() {
+    let isValid = true, isCompleted = false;
+    isValid = this.dateIsValid();
+    isCompleted = !!this.state.value.length;
+    this.setState({ isValid: isValid, completed: isCompleted });
+    return !!this.props.wrapper.required ? isValid && isCompleted : isValid;
+  }
+
+  buildValue() {
+    return this.state.month + "-" + this.state.day + "-" + this.state.year;
```

Comment:
In cases like this, I prefer this syntax to backticks/templates. But that's just a personal preference.